pkg/synchronization/controller.go

```diff
@@ -133,14 +136,10 @@ func (c *Controller) Process(ctx *ctrl.ProcessContext) (bool /* retriable */, er
 	}
 
 	ctx.Logger.Sugar().Infof("Looking up service data for service %q from Service Central", serviceName)
-	serviceData, err := c.getServiceData(auth.NoUser(), serviceName)
-	if err != nil {
-		if servicecentral.IsNotFound(err) {
-			// no retries if SC is missing a service record
-			return false, err
-		}
-		// should be able to retry other SC errors
-		return true, err
+	serviceData, ok := c.getCachedServiceData(serviceName)
+	if !ok {
+		// no retries if service is missing in cache
+		return false, errors.New("service not found in cache")
```

Comment:
Should this not try to fetch if the service is not there? The only reason it's not there is on initial service create right?

Reply:
@ychen-atlassian actually, even on creation it shouldn't be possible to run into this, because user won't have permissions to create SD until we create a ClusterRoleBinding for SD, which requires fetching (and caching) data from Service Central.

So really this should never happen, hence no need to fetch.